cloudbridge/cloud/providers/gce/resources.py

```diff
@@ -828,8 +828,8 @@ def instance_type_id(self):
         machine_type_uri = self._gce_instance.get('machineType')
         if machine_type_uri is None:
             return None
-        instance_type = self._provider.get_gce_resource_data(machine_type_uri)
-        return instance_type.get('name', None)
+        parsed_uri = self._provider.parse_url(machine_type_uri)
```

Comment:
It would be great to have an example url in the GCPResource.parse_url() method docs so you can see what kind of URL is being split apart, and what the split parameters look like.

In addition, an example return dict in the method docs for GCPResourceURL.get() would also be helpful to understand the code without further referencing.

About the TODO item in GCPResources. I can see why there's a need to access the protected member in connection._resourceDesc in GCPResources - it seems a lot more efficient than hitting the endpoint again to retrieve resource definitions. If you think it's unlikely to change in the near future, I think we may as well leave it as is, with the TODO changing to a "FIX_IF_NEEDED". Boto3 has a similar mechanism, however, boto3's resource definitions/metadata is public, so you can access it safely.

Reply:
Done